python/example_code/ses/ses_replicateidentities/ses_replicateidentities.py

```diff
@@ -0,0 +1,343 @@
+            except ClientError:
+                print("Invalid ARN")
+    elif ask == 'no':
+        return
+
+
+# Main
+SRC_REGION = input("Which region do you want to replicate from? ")
+if SRC_REGION in regions:
+    ses_source_client = S.client('ses', region_name=SRC_REGION)
+    region_email_identities = []
+    region_dom_identities = []
+    # Added pagination for listing SES identities.
+    paginator = ses_source_client.get_paginator('list_identities')
+    response_iterator = paginator.paginate(PaginationConfig={
+                                                             'MaxItems': 20,
+                                                             'PageSize': 20
+                                                            }
+                                           )
+    for x in response_iterator:
+        for y in x['Identities']:
+            if '@' in y:
+                region_email_identities.append(y)
+            else:
+                region_dom_identities.append(y)
+    print ("Email addresses in source region:")
+    print(region_email_identities)
+    print ("Domains in source region:")
+    print(region_dom_identities)
+
+else:
+    sys.exit("Region entered invalid. \
+             Please enter a region where SES is available.")
+DST_REGION = input("Which region do you want to replicate to? ")
+if DST_REGION in regions:
+    email_call = email_verify(region_email_identities, DST_REGION)
+    verification_table = domain_verify(region_dom_identities, DST_REGION)
+    # Route53 subroutine. Based on user-input, used if
+    # domains being verified are in Route53.
+    r53dom = input("Are your domains hosted in Route53?(yes/no) ")
+    if r53dom == 'yes':
+        # Prints domain names and their verification tokens.
+        print(verification_table)
+        print("")
+        add_route_53_record(table=verification_table, rec_type='domainVerify')
+    elif r53dom == 'no':
+        print("Use the verification tokens returned to ",
+              "create TXT records through your DNS provider.")
+        # Prints domain names and their verification tokens.
+        print(verification_table)
+        print("")
+
+    # SNS topic addition
+    sns = input("Do you want to add SNS notifications "
+                "for the identities? (yes/no) ")
+    if sns == 'yes':
+        for addr in region_email_identities:
+            sns_topics(addr, DST_REGION)
+        for dom in region_dom_identities:
+            sns_topics(dom, DST_REGION)
+    elif sns == 'no':
+        pass
+
+    # DKIM verification
+    dkim = input("Do you want to configure DKIM for the identities? (yes/no) ")
+    if dkim == 'yes':
+        for addr in region_email_identities:
+            generate_dkim(addr, DST_REGION)
+        for dom in region_dom_identities:
+            generate_dkim(dom, DST_REGION)
+
+else:
+    sys.exit("Region entered invalid. \
+             Please enter a region where SES is available.")
```

Comment:
Collect this into a main() function that's called only when the script is invoked directly.

Reply:
This is also done.